src/test/scala/nl.knaw.dans.easy.deposit/docs/DatasetXmlSpec.scala

```diff
@@ -22,22 +22,73 @@ import better.files.File
 import javax.xml.XMLConstants
 import javax.xml.transform.Source
 import javax.xml.transform.stream.StreamSource
-import javax.xml.validation.SchemaFactory
+import javax.xml.validation.{ Schema, SchemaFactory }
 import nl.knaw.dans.easy.deposit.TestSupportFixture
 import nl.knaw.dans.easy.deposit.docs.DatasetMetadata.{ DateQualifier, QualifiedSchemedValue }
 import resource.Using
 
 import scala.util.{ Failure, Success, Try }
-import scala.xml.{ Node, Utility }
+import scala.xml.{ Elem, Node, PrettyPrinter, Utility }
 
-class DatasetXmlSpec extends TestSupportFixture {
+trait DdmBehavior {
+  this: TestSupportFixture =>
 
-  private lazy val triedSchema = Try { // loading postponed until we actually start validating
+  lazy val triedSchema: Try[Schema] = Try { // loading postponed until we actually start validating
```

Comment:
you're doing something with `getOrElse` and `fail` below. Would it make sense to apply that here as well?